### PR TITLE
Update protobuf to 3.9.0

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @TeBoring isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

File: grpc.gemspec [hunk 1]
```diff
@@ -29,7 +29,7 @@ Gem::Specification.new do |s|
   s.require_paths = %w( src/ruby/lib src/ruby/bin src/ruby/pb )
   s.platform      = Gem::Platform::RUBY
 
-  s.add_dependency 'google-protobuf', '~> 3.8'
+  s.add_dependency 'google-protobuf', '~> 3.9.0'
```
Comment (jtattermusch) [inline grpc.gemspec]: I think the missing `.0` is intended. (see PR for bump to 3.8). Please fix in the .template file

**Comment by jtattermusch** [comment]
It is useful to provide link to the last update https://github.com/grpc/grpc/pull/19182 - most changes should be of the same nature.

**Comment by jtattermusch** [comment]
@TeBoring  for some reason the CLA check is failing - can you doublecheck please?

**Comment by TeBoring** [comment]
What do I need to do for the CLA? Can we sign the cla for the foundation?

**Comment by jtattermusch** [comment]
CLA: https://identity.linuxfoundation.org/projects/cncf  you should be able to sign as google employee.

**Comment by jtattermusch** [comment]
The c# error is something we need to take a look into. It looks like 3.9.0 might contain a breaking change that needs to be rolled back.
```
  Grpc.Microbenchmarks -> /var/local/git/grpc/src/csharp/Grpc.Microbenchmarks/bin/Debug/net461/Grpc.Microbenchmarks.exe
Math.cs(39,17): error CS1729: 'GeneratedClrTypeInfo' does not contain a constructor that takes 7 arguments [/var/local/git/grpc/src/csharp/Grpc.Examples/Grpc.Examples.csproj]
Math.cs(40,17): error CS1729: 'GeneratedClrTypeInfo' does not contain a constructor that takes 7 arguments [/var/local/git/grpc/src/csharp/Grpc.Examples/Grpc.Examples.csproj]
Math.cs(41,17): error CS1729: 'GeneratedClrTypeInfo' does not contain a constructor that takes 7 arguments [/var/local/git/grpc/src/csharp/Grpc.Examples/Grpc.Examples.csproj]
Math.cs(42,17): error CS1729: 'GeneratedClrTypeInfo' does not contain a constructor that takes 7 arguments [/var/local/git/grpc/src/csharp/Grpc.Examples/Grpc.Examples.csproj]
Math.cs(43,17): error CS1729: 'GeneratedClrTypeInfo' does not contain a constructor that takes 7 arguments [/var/local/git/grpc/src/csharp/Grpc.Examples/Grpc.Examples.csproj]
```

Other problems:
https://source.cloud.google.com/results/invocations/c03ae785-c291-4a23-95a9-c0b8a91decf6/targets/github%2Fgrpc%2Fsanity_linux_dbg_native/tests

**Comment by jtattermusch** [comment]
The C# problem likely has to do with this change:
https://github.com/protocolbuffers/protobuf/commit/9e89b6e70a4988a4b473a19a7647154bd07583ce#diff-a11488f894c3a2f0a362c3ea33941a88R107

**Comment by jtattermusch** [comment]
CC @ObsidianMinor  (probably related to https://github.com/protocolbuffers/protobuf/pull/5350)

**Comment by TeBoring** [comment]
I signed

**Comment by ObsidianMinor** [comment]
[3.9.0 was delisted](https://github.com/protocolbuffers/protobuf/issues/6379#issuecomment-511946613) until a 3.9.1 release can be made to with [these changes](https://github.com/protocolbuffers/protobuf/pull/6380). The C# issues are probably related to that.

**Comment by anandolee** [comment]
Looks like the test is using new protoc but old c# runtime?  @jtattermusch is the failure in this case, and why old c# runtime is used in Grpc's test?

Protobuf only gurantee old generated code work with new runtime. Not gurantee new generated code work with old runtime

**Comment by jtattermusch** [comment]
Ok, I think I now understand what happened:
- this PR attempts to increase C# protobuf runtime version in Version.csproj.include, but it turns out that's a dead copy that doesn't influence anything, so the protobuf runtime is not really updated, but the code is regenerated. That explains the C# build error. 

I created https://github.com/grpc/grpc/pull/19679 to fix the problem. After it's merged, this PR needs to be updated to updates `src/csharp/build/dependencies.props` instead of `Version.csproj.include`

**Pull request closed without merging**